### PR TITLE
Making `open` case-insensitive to file extensions

### LINKED ISSUE #10441: open case-sensitive to file extensions on Windows
### Describe the bug

When opening a file that nushell automatically parses into an object (tsv, json, etc) the open command is case-sensitive to the file extension on Windows.

### How to reproduce

Create two identical files, one with an extension like ".JSON" and the other with the regular ".json". Opening the ".json" file will give the expected data structure. Opening the ".JSON" file will give the raw text. 

### Expected behavior

At least on windows, and justifiably on unix for some situations, the case of the file extension should not matter. My own experience is with biological data, and lots of different programs/data sources choose to express the file extension differently. A fasta and a Fasta and a FASTA should all parse the same, and an mzML and an MZML should parse the same. The same should be true for built-in file types. 

### Screenshots


![image](https://github.com/nushell/nushell/assets/70269632/51052d5e-479b-4c94-bd5d-7f6b17cd82e5)



### Configuration

| key                | value                                                                                                                                                                                                                                                                                                |
| ------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| version            | 0.84.0                                                                                                                                                                                                                                                                                               |
| branch             |                                                                                                                                                                                                                                                                                                      |
| commit_hash        | d2abb8603abab43ec4a6d4ed56f9018790b99dce                                                                                                                                                                                                                                                             |
| build_os           | windows-x86_64                                                                                                                                                                                                                                                                                       |
| build_target       | x86_64-pc-windows-msvc |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16) |
| rust_channel       | 1.69.0-x86_64-pc-windows-msvc |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12) |
| build_time         | 2023-08-22 21:21:31 +00:00 |
| build_rust_channel | release |
| allocator          | standard |
| features           | default, sqlite, trash, which, zip |
| installed_plugins  | from bam, from bcf, from bcf.gz, from bed, from cram, from eml, from fa, from fa.gz, from fasta, from fasta.gz, from fastq, from fastq.gz, from fq, from fq.gz, from gfa, from gfa.gz, from gff, from ics, from ini, from sam, from vcf, from vcf.gz, query, query json, query web, query xml, regex |
### Additional context

_No response_

### PR DESCRIPTION
**Pull request by poketch**
# Description

Closes #10441 

Uses `String::to_lowercase()` when the file's extension `ext` is parsed to allow `from_decl(format!("from {ext}"))` to return the desired output regardless of extension case. 

It doesn't work with sqlite files since those are handled earlier in the parsing but I think is good- since there's no standard file extension used by sqlite so a user will likely want case sensitivity in that case.

This also has the (possibly undesired) effect of making `open` completely case insensitive, e.g. `open foo.JSON` will work on a file named `foo.json` and vice versa. This is good on Windows as it treats `foo.json` and `foo.JSON` as the same file, but may not be the desired behaviour on Unix.

If this behaviour is undesired I assume it would be fixed with a `#[cfg(not(unix))]` attribute on the `to_lowercase()` operation but that produces slightly "uglier" code that I didn't wish to submit unless necessary. 

old behaviour:
image

new behaviour: 
image

# User-Facing Changes

`open` will now present a table when `open`-ing files with captitalized extensions rather than the file's raw data

# Tests + Formatting

new test: `parses_file_with_uppercase_extension` which tests the desired behaviour

### CODE REVIEW

File: crates/nu-command/tests/commands/open.rs [hunk 1]
```diff
@@ -65,6 +96,17 @@ fn parses_bson() {
     assert_eq!(actual.out, "hello");
 }
 
+#[cfg(feature = "bson")]
+#[test]
+fn parses_bson() {
+    let actual = nu!(
+        cwd: "tests/fixtures/formats",
+        "open sample.bson | get root | select 0 | get b"
```
Comment (jntrnr) [inline crates/nu-command/tests/commands/open.rs:104]: Not sure how this test case relates to the fix?

**Comment by jntrnr** [comment]
Looks good. Other than what seems like an unrelated test into the mix, looks ready to land.

File: crates/nu-command/tests/commands/open.rs (hunk 1 above)
Comment (poketch) [reply crates/nu-command/tests/commands/open.rs:104]: ah that's an artifact from when I thought I needed to duplicate all the tests. I'll remove it

**Comment by sholderbach** [comment]
Thanks for the quick turnaround. One caveat is that the command names are only by convention lowercase. So a command with weird capitalization would be blocked from both an exact match or case insensitive match in this setup. 
As all nushell internal commands are following this standard and we could document it, I can be convinced to forgo the complication of making it completely case insensitive (with `.find_decl()` etc. that is more involved)

**Comment by poketch** [comment]
I'll defer to your judgement but striving for consistent capitaliszation where possible seems good to me. I can update any relevant docmuentation where necessary as part of this PR (though I'll need a point in the right direction).

**Comment by amtoine** [comment]
do we need to do that when one can simply implement the following and have `open` do the work? :thinking: 
```nushell
def "from JSON" [] { from json }
```

this is the same idea behing the `std formats "from ndjson"` command in the standard library :yum: 
https://github.com/nushell/nushell/blob/addf8ca942a85e6a061e8315afbe90c95ae4b3e4/crates/nu-std/std/formats.nu#L12-L15

**Comment by sholderbach** [comment]
> I'll defer to your judgement but striving for consistent capitaliszation where possible seems good to me. I can update any relevant docmuentation where necessary as part of this PR (though I'll need a point in the right direction).

The documentation of `open`'s command lookup we have right now was only recently defined in #10255 and https://github.com/nushell/nushell.github.io/pull/1048. So that may be the prime locations for documenting (plus possibly the stub command for `from`

> do we need to do that when one can simply implement the following and have `open` do the work? 🤔
> 
> ```nushell
> def "from JSON" [] { from json }
> ```

Downside to defining a bunch of aliasing new commands is that they pollute the scope and would appear in the `help`

**Comment by amtoine** [comment]
> Downside to defining a bunch of aliasing new commands is that they pollute the scope and would appear in the `help`
>
good point :+1: 

i can leave with both, the JSON from `open` or the `"from json"` defined outside :+1:

**Comment by sholderbach** [comment]
Let's get this QoL improvement in as is. Thanks for the PR @poketch !

**Pull request merged**